includes/class-wc-payment-tokens.php

```diff
@@ -0,0 +1,249 @@
+		$wpdb->insert( $wpdb->prefix . 'woocommerce_payment_tokens', $core_fields_to_create );
+		$token_id = $wpdb->insert_id;
+
+		// Insert any other meta and associate it with the new token
+		foreach ( $args as $meta_key => $meta_value ) {
+			$wpdb->insert( $wpdb->prefix . 'woocommerce_payment_token_meta', array(
+				'meta_key'   => $meta_key,
+				'meta_value' => $meta_value,
+				'token_id'   => $token_id
+			) );
+		}
+
+		do_action( 'woocommerce_payment_token_created', $token_id );
+		return self::generate_token( $token_id, $core_fields_to_create );
+	}
+
+	/**
+	 * Updates (stores in the database), and returns a payment token object
+	 * @param int               $token_id Token ID
+	 * @param  WC_Payment_Token $token    An existing payment token object (updated fields can be changed with set_)
+	 * @return WC_Payment_Token Updated token
+	 */
+	public static function update( $token_id, $token ) {
+		$args = $token->__data_format();
+		unset( $args['token_id'] );
+		if ( ! self::validate( $args ) ) {
+			return; // @todo throw an error
+		}
+
+		// We need to separate our meta fields since they are stored in a separate table
+		$core_fields = self::get_token_core_fields();
+		$core_fields_to_create = array();
+		foreach ( $core_fields as $core_field ) {
+			if ( isset( $args[ $core_field ] ) ) {
+				$core_fields_to_update[ $core_field ] = $args[ $core_field ];
+				unset( $args[ $core_field ] );
+			}
+		}
+
+		global $wpdb;
+
+		// Update our core fields
+		$wpdb->update( $wpdb->prefix . 'woocommerce_payment_tokens', $core_fields_to_update, array( 'token_id' => $token_id ) );
+
+		// Grab a list of our existing meta, and update if we have a matching value in $args, otherise, create a new meta row
+		$existing_meta = $wpdb->get_col( $wpdb->prepare(
+			"SELECT meta_key FROM {$wpdb->prefix}woocommerce_payment_token_meta WHERE token_id = %d", $token_id
+		) );
+		foreach ( $args as $meta_key => $meta_value ) {
+			if ( in_array( $meta_key, $existing_meta ) ) {
+				$wpdb->update( $wpdb->prefix . 'woocommerce_payment_token_meta', array(
+					'meta_value' => $meta_value
+				), array( 'token_id'   => $token_id, 'meta_key'   => $meta_key ) );
+			} else {
+				$wpdb->insert( $wpdb->prefix . 'woocommerce_payment_token_meta', array(
+					'meta_key'   => $meta_key,
+					'meta_value' => $meta_value,
+					'token_id'   => $token_id
+				) );
+			}
+		}
+
+		do_action( 'woocommerce_payment_token_updated', $token_id );
+		return self::generate_token( $token_id, $core_fields_to_update );
+	}
+
+	/**
+	 * Remove a payment token from the database
+	 * @param int $token_id Token ID
+	 */
+	public static function delete( $token_id ) {
+		global $wpdb;
+		$wpdb->delete( $wpdb->prefix . 'woocommerce_payment_tokens', array( 'token_id' => $token_id ), array( '%d' ) );
+		$wpdb->delete( $wpdb->prefix . 'woocommerce_payment_token_meta', array( 'token_id' => $token_id ), array( '%d' ) );
+		do_action( 'woocommerce_payment_token_deleted', $token_id );
```

Comment:
might be worth retrieving the token object and passing it as an arg here, as functions hooking in here that need more than just the token ID would have no way to get that information at this point.